Api automation 1.0

RafaDdS opened the pull request:
Creates an automation system to implement the bindings based on an input file. 
Most of the Core and ImgProc have been re-implemented using the system.
Some of the tests for the new methods.
Updated to Godot 4.3 to use in editor documentation.

matt-s-clark commented on SConstruct:
To compile on MacOS, line 18 of SConstruct needs to be commented out. I wonder if there is an easy way to include/remove based on arch?

RafaDdS replied:
I could set it to be included based on arch, but it's not needed on Linux. if it's also not important to Windows we could just remove it.

matt-s-clark commented on demo/Examples/CascadeClassifierExample.gd:
This used to recognise my fist, but I had trouble getting the classifer to draw a square on my fist this time around. The square (overlay) did pop up in a couple of places, but not on my fist.

matt-s-clark reviewed (commented):
For MacOS, GUT did crash when attempting to run all of the tests. 

Test that crashes GUT:
* test_video_capture.gd

Test that failed with "Out of bounds get index'0' (on base: 'Array'):
* test_face.gd

Test that failed with "Invalid get index 'x' (on base 'Nil')"
* test_trackers.gd

Test that passed 3 items:
* test_video_capture.gd

I don't think there are any show stoppers there. Great work!

RafaDdS committed:
Adding Checks for null and empty values

RafaDdS committed:
Casting Video Capture open to int

RafaDdS commented:
I added checks to give better test outputs and made a small change that might solve the crash.

RafaDdS committed:
Simplifying OpenCV Install and creating CI for Linux (#13)

* Adding opencv as submodule

* Update SConstruct

- Added link flags to fail build in case of missing objects
- Added dnn library
- Cleaned library and header paths to use only the opencv build as dependency

* OpenCV Setup bash

- Removing submodules

* Version correction

- Ignoring OpenCV source code

* Update build.yml

* CI test

* CI test

* Changing OpenCV Path

RafaDdS committed:
Merge branch 'master' into api-automation
